docs/Tutorial_DAMP.ipynb

```diff
@@ -0,0 +1,778 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #174.            if stop_idx &gt;= excl_zone_st:
</pre><p>I actually think it's better to write <code>if stop &gt;= query_idx - excl_zone</code> because it's very clear that you want to stop before you get to the <code>query_idx</code> and, if the exclusion zone is greater than zero then make sure <code>stop</code> does not go past that barrier either.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/872/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<blockquote>&nbsp;actually think it's better to write</blockquote><p>Just to make sure that I am not missing anything here:</p><p>Are you suggesting that I should avoid using the intermediate variable <code>excl_zone_st</code>and just write <code>query_idx - excl_zone</code>  in the inequality ?</p><p> </p><p> </p><p> </p><p> </p><blockquote>&nbsp;if the exclusion zone is greater than zero then make sure&nbsp;<code>stop</code>&nbsp;does not go past that barrier either.</blockquote><p> </p><p>I think it all depends on how we want to definte "stop". Is "stop" the stop index of a time series OR is it the stop index before which the last subsequence of the time series starts? </p><p> </p><p>e.g. <code>T = [0, 1, 2, 3, 4, 5]; m=3</code>  </p><p> </p><p>The "stop" is <code>6</code> or <code>4</code>? </p><p> </p><p>A non-trivial neighbor is a subsequence whose start index is outside of the exclusion zone. So, If I define "stop" as the stop index of the time series (e.g. <code>6</code>  in example above) then I am okay if <code>stop</code> goes past the barrier as long as the start index of that subsequence is before the barries. </p><p> </p><p>Please allow me to revise the code. Your comment, this long explanation (a red flag) as well as your next comment regarding line 175 all indicate that my code is not clear enough. I will try to clean it.</p>

Reply:
<blockquote>Are you suggesting that I should avoid using the intermediate variable&nbsp;<code>excl_zone_st</code>and just write&nbsp;<code>query_idx - excl_zone</code>&nbsp;in the inequality ?</blockquote><p>Yes, I find <code>excl_zone_st</code> to be ugly :(</p><p> </p><blockquote>Please allow me to revise the code. Your comment, this long explanation (a red flag) as well as your next comment regarding line 175 all indicate that my code is not clear enough. I will try to clean it.</blockquote><p>Thank you! I'm really glad that you got my point. It will be very hard to maintain and reason with in the future if we don't get this right</p><p> </p>

Reply:
> It will be very hard to maintain and reason with in the future if we don't get this right. I don't doubt that the code "works" but it needs to be more obvious and also consistent (regarding what `start` and `stop` refer to)

I totally understand. I checked the MERLIN notebook a few days ago and it was not easy for me to follow the code despite the fact that I wrote the code myself. So, I do not want to upset my future self this time :)